### PR TITLE
fixes clean shutdown bug in manager

### LINKED ISSUE #4209: Elasticity Broken or Flaky test: ReadWriteIT.sunnyDay()
**Test name(s)**
 - ReadWriteIT.sunnyDay()

**Describe the failure observed**
The test does not always pass. The manager lock is sometimes held until the test times out.

The message logged until timeout: `[functional.ReadWriteIT] INFO : Manager lock is still held`
The error the test fails with: `org.opentest4j.AssertionFailedError: execution timed out after 360000 ms`

Error is occurring in the elasticity branch

### PR DESCRIPTION
**Pull request by keith-turner**
This commits fixes #4209 by shutting down fate before unassigning any tablets on which fate depends.

### CODE REVIEW

**Review by cshannon — COMMENTED** [review]
I took a look at this and my main thought is I don't think this solves the problem all the way because what happens if the Tservers are all offline or can't be reached due to something like a network issue?

I think this may only help in the case that the Manager is shutting things down.

**Comment by keith-turner** [comment]
> I took a look at this and my main thought is I don't think this solves the problem all the way because what happens if the Tservers are all offline or can't be reached due to something like a network issue?

Made an update in [5e47c1a](https://github.com/apache/accumulo/pull/4248/commits/5e47c1a0851db7f042e94552067cb6df67f63862) to not wait forever.  Realized in addition to tservers that other things like DFS and ZK could make fate operations running in the background threads hang. So the changes wait a bit, giving thing running in background threads a chance to wind down.

File: core/src/main/java/org/apache/accumulo/core/fate/Fate.java [hunk 1]
```diff
@@ -458,10 +458,14 @@ public Exception getException(FateId fateId) {
...
+  }
+
   /**
    * Initiates shutdown of background threads and optionally waits on them.
    */
-  public void shutdown(boolean waitForBackgroundThreads) {
+  public void shutdown(long timeout, TimeUnit timeUnit) {
     if (keepRunning.compareAndSet(true, false)) {
       fatePoolWatcher.shutdown();
       if (executor != null) {
```
Comment (cshannon) [inline core/src/main/java/org/apache/accumulo/core/fate/Fate.java:471]: This is unrelated to your PR but something I noticed is executor should never be null as it's final and created in the constructor.

Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/fate/Fate.java]: did not realize that.  Removed the null checks and it cleaned up the code.

Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/fate/Fate.java]: [fd5b2a5](https://github.com/apache/accumulo/pull/4248/commits/fd5b2a5d22ab71a9195b5f1aba80ae8201be836e)

**Review by cshannon — APPROVED** [review]
LGTM now, I tested ReadWriteIT a few times and it no longer hangs.

**Pull request merged**